### ISSUE TITLE
Sending in unexpected referenceId reads/search request causes 500

### ISSUE THREAD

**Issue by david4096** [ga4gh/server#1050]
When sending in requests to `reads/search` if it can't find the reference it fails with an Internal Server Error.

```
➜  server-1 git:(master) ✗ http post kozbo-vm.cloudapp.net/reads/search readGroupIds:='["SEcwMDA5OTpIRzAwMDk5Lm1hcHBlZC5JTExVTUlOQS5id2EuR0JSLmxvd19jb3ZlcmFnZS4yMDEzMDQxNTpTUlI3NDE0MTE"]' referenceId=1  
HTTP/1.0 500 INTERNAL SERVER ERROR
Content-Length: 61
Content-Type: application/json
Date: Sat, 02 Apr 2016 00:32:09 GMT
Server: Werkzeug/0.11.5 Python/2.7.6

{
    "errorCode": 1277500761, 
    "message": "Internal Server Error"
}
```

**Comment by kozbo** [comment]
Here is the output on the server side:
```
ERROR in app [/srv/ga4gh/ga4gh-server-env/local/lib/python2.7/site-packages/flask/app.py:1423]:
Exception on /reads/search [POST]
--------------------------------------------------------------------------------
Traceback (most recent call last):
  File "/srv/ga4gh/ga4gh-server-env/local/lib/python2.7/site-packages/flask/app.py", line 1475, in full_dispatch_request
    rv = self.dispatch_request()
  File "/srv/ga4gh/ga4gh-server-env/local/lib/python2.7/site-packages/flask/app.py", line 1461, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "/srv/ga4gh/ga4gh-server-env/local/lib/python2.7/site-packages/ga4gh-0.2.0a2.dev170+ngcce89bc-py2.7.egg/ga4gh/frontend.py", line 483, in searchReads
    flask.request, app.backend.runSearchReads)
  File "/srv/ga4gh/ga4gh-server-env/local/lib/python2.7/site-packages/ga4gh-0.2.0a2.dev170+ngcce89bc-py2.7.egg/ga4gh/frontend.py", line 407, in handleFlaskPostRequest
    return handleHttpPost(flaskRequest, endpoint)
  File "/srv/ga4gh/ga4gh-server-env/local/lib/python2.7/site-packages/ga4gh-0.2.0a2.dev170+ngcce89bc-py2.7.egg/ga4gh/frontend.py", line 284, in handleHttpPost
    responseStr = endpoint(request.get_data())
  File "/srv/ga4gh/ga4gh-server-env/local/lib/python2.7/site-packages/ga4gh-0.2.0a2.dev170+ngcce89bc-py2.7.egg/ga4gh/backend.py", line 760, in runSearchReads
    self.readsGenerator)
  File "/srv/ga4gh/ga4gh-server-env/local/lib/python2.7/site-packages/ga4gh-0.2.0a2.dev170+ngcce89bc-py2.7.egg/ga4gh/backend.py", line 616, in runSearchRequest
    for obj, nextPageToken in objectGenerator(request):
  File "/srv/ga4gh/ga4gh-server-env/local/lib/python2.7/site-packages/ga4gh-0.2.0a2.dev170+ngcce89bc-py2.7.egg/ga4gh/backend.py", line 526, in readsGenerator
    reference = referenceSet.getReference(request.referenceId)
AttributeError: 'NoneType' object has no attribute 'getReference'
169.233.227.208 - - [02/Apr/2016 00:32:09] "POST /reads/search HTTP/1.1" 500 -
```

**Comment by dcolligan** [comment]
Can't reproduce.  Can you put an `import pdb; pdb.set_trace()` somewhere and dump the request that the server is receiving?

Also what is the `http` utility you are using?

These lines in the stacktrace are helpful:
```
    reference = referenceSet.getReference(request.referenceId)
AttributeError: 'NoneType' object has no attribute 'getReference'
```

They indicate that the issue is that a readGroupSet did not have a referenceSet assigned to it, rather than a lookup failure for a reference (which returns a 404, correctly).

**Comment by david4096** [comment]
To replicate I generated the compliance dataset using `prepare_compliance_data`. I then added the referenceSet from @jeromekelleher's [example data](http://www.well.ox.ac.uk/%7Ejk/ga4gh-example-data-v3.0.tar).

[HTTPie](https://github.com/jkbrzt/httpie) is a user friendly curl replacement. The request I sent is equivalent to `curl -X POST -H "Content-Type: application/json" -d '{"readGroupIds": ["YnJjYTE6SEcwMDA5NjpTUlIwNjI2MzQ"], "referenceId": "1"}' http://localhost:8000/reads/search`.

**Comment by dcolligan** [comment]
Yeah, this is a data problem primarily; I couldn't reproduce the issue (using a valid readGroupId I have lying around):

Using httpie:
```
$ http post localhost:8000/reads/search readGroupIds:='["MWtnLXAzLXN1YnNldDpIRzAwMDk2OlNSUjA2MjYzNA"]' referenceId=1
HTTP/1.0 404 NOT FOUND
Content-Length: 65
Content-Type: application/json
Date: Tue, 05 Apr 2016 15:15:48 GMT
Server: Werkzeug/0.11.5 Python/2.7.5

{
    "errorCode": 1763742139,
    "message": "referenceId '1' not found"
}
```

Using curl:
```
$ curl -X POST -H "Content-Type: application/json" -d '{"readGroupIds": ["MWtnLXAzLXN1YnNldDpIRzAwMDk2OlNSUjA2MjYzNA"], "referenceId": "1"}' http://localhost:8000/reads/search
{"errorCode": 1763742139, "message": "referenceId '1' not found"}
```

The readGroupSet that you are using in your data set doesn't have a reference set assigned to it.

**Comment by david4096** [comment]
To replicate:

```
python scripts/prepare_compliance_data.py -o ga4gh-example-data
wget http://www.well.ox.ac.uk/%7Ejk/ga4gh-example-data-v3.0.tar
tar -xf ga4gh-example-data-v3.0.tar
python repo_dev.py remove-referenceset ga4gh-example-data hg37
python server_dev.py
curl -X POST -H "Content-Type: application/json" -d '{"readGroupIds": ["YnJjYTE6SEcwMDA5NjpTUlIwNjI2MzQ"], "referenceId": "1"}' http://localhost:8000/reads/search
```

**Comment by dcolligan** [comment]
I can reproduce the error.  So what should be the expected behavior here?

**Comment by david4096** [comment]
I think it should produce a 400 bad request if the reference ID isn't found.

**Comment by dcolligan** [comment]
Fixed in #1081
